standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java

```diff
@@ -1251,4 +1251,19 @@ public static String getHttpPath(String httpPath) {
     }
     return httpPath;
   }
+
+  /**
+   * Function to get the base name of the column type.
+   * @param colType column type
+   * @return base name of the column type
+   */
+  public static String getBaseNameOfColType(String colType) {
+    // char, varchar types can have parameters such as
+    // char/varchar length. We only need the base name.
+    int idx = colType.indexOf('(');
+    if (idx != -1) {
+      return colType.substring(0, idx);
```

Comment:
Is coltype already trimmed here? what if there are whitespaces at the end? 
For example `varchar(30)  `
If this could be an issue, a test case also should be added in `TestHiveMetaStore.testPartitionFilter()`